### ISSUE TITLE
aqua update-aqua failed on Windows

### ISSUE THREAD

**Issue by suzuki-shunsuke** [aquaproj/aqua#2144]
### aqua info

```console
$ aqua -v
aqua version 2.10.1 (fbf3b157b21897fa2f315fbc30474acafded108d)
```

OS (Windows, Linux, macOS, etc): windows
CPU Architecture (amd64, arm64, etc): amd64

### Overview

`aqua update-aqua` failed on Windows.

### How to reproduce

Run `aqua update-aqua` with the latest aqua on Windows.

### Expected behaviour

aqua skips updating aqua because the current aqua version is same with the new version.

### Actual behaviour

aqua tried to update aqua and failed.

https://github.com/aquaproj/aqua/actions/runs/5704029804/job/15457238499

```
FATA[0000] aqua failed                                   aqua_version=2.10.1 env=windows/amd64 error="download aqua: create a file: open C:\\Users\\runneradmin\\AppData\\Local\\aquaproj-aqua\\bin\\aqua.exe: The process cannot access the file because it is being used by another process." new_version=v2.10.1 program=aqua
Error: Process completed with exit code 1.
```

### Important Factoids

`aqua_version` is not `v2.10.1` but `2.10.1`.

**Comment by airtonix** [comment]
so my experiences is slightly different. 

1. be on v2.21.2
2. run `aqua update-aqua`
3. result: nothing. no output.
4. run `aqua update-aqua v2.17.4`
5. result: nothing. no output.
6. run `aqua update-aqua 2.17.4`
7. result: get github error that it can't find the `2.17.4` tag


error `#7` makes sense since the releases are tagged in the format `v{version}`

**Comment by suzuki-shunsuke** [comment]
https://github.com/orgs/aquaproj/discussions/2510#discussioncomment-7647449

**Comment by suzuki-shunsuke** [comment]
> run aqua update-aqua
> result: nothing. no output.

`aqua update-aqua` doesn't output anything if the new version is already installed.

**Comment by suzuki-shunsuke** [comment]
There are several solutions.
We'll try them.

https://stackoverflow.com/questions/1211948/best-method-for-implementing-self-updating-software

**Comment by airtonix** [comment]
@suzuki-shunsuke Are you saying that your vision for `aqua update-aqua` is that it only ever installs the latest version? 

clarifying that this mean it won't ever install versions older than the bootstrap version? 

if this is the case, then we shouldn't bother accepting a version parameter and make it clear that the installer only ever installs the latest version possible.

**Comment by suzuki-shunsuke** [comment]
> Are you saying that your vision for aqua update-aqua is that it only ever installs the latest version?

No. `aqua update-aqua` installs the latest version by default, but you can install old versions.

```sh
aqua update-aqua [<version>]
```

e.g.

```sh
aqua update-aqua v2.0.0
```
